packages/process/process_monitor.pony

```diff
@@ -183,34 +113,22 @@ type ProcessMonitorAuth is (AmbientAuth | StartProcessAuth)
 
 actor ProcessMonitor
   """
-  Forks and monitors a process. Notifies a client about STDOUT / STDERR events.
+  Fork+execs / creates a child process and monitors it. Notifies a client about STDOUT / STDERR events.
   """
   let _notifier: ProcessNotify
-  var _child_pid: I32 = -1
-
   let _backpressure_auth: BackpressureAuth
 
-  var _stdin_event: AsioEventID = AsioEvent.none()
-  var _stdout_event: AsioEventID = AsioEvent.none()
-  var _stderr_event: AsioEventID = AsioEvent.none()
+  var _stdin: _Pipe = _Pipe.none()
+  var _stdout: _Pipe = _Pipe.none()
+  var _stderr: _Pipe = _Pipe.none()
+  var _child: _Process = _ProcessNone
 
   let _max_size: USize = 4096
   var _read_buf: Array[U8] iso = recover Array[U8] .> undefined(_max_size) end
   var _read_len: USize = 0
   var _expect: USize = 0
-  embed _pending: List[(ByteSeq, USize)] = _pending.create()
-
-  var _stdin_read: U32 = -1
-  var _stdin_write: U32 = -1
-  var _stdout_read: U32 = -1
-  var _stdout_write: U32 = -1
-  var _stderr_read: U32 = -1
-  var _stderr_write: U32 = -1
-
-  var _stdin_writeable: Bool = true
-  var _stdout_readable: Bool = true
-  var _stderr_readable: Bool = true
```

Comment:
These writable and readable flags aren't really needed. `_stdin_writeable` was not always set, and was redundant with `_pending` being non-empty. `_std*_readable` was set, but never read. So now using just the state of the pipes plus the state of the buffers.